### DISCUSSION TITLE
Terminal.Gui in devcontainers in vscode?

### DISCUSSION THREAD

**Discussion by thomaslazar** [tui-cs/Terminal.Gui#3049] [Q&A]
Hi there. 

I was wondering if someone already got the whole Terminal.Gui stuff running in a devcontainer in vscode. 

I'm doing a bit of playing around with some stuff and wanted to quickly build a small UI for it, but I can't get Terminal.Gui projects to run inside a dotnet devcontainer. 

Always getting a `Error opening terminal: unknown.` in the end. 

Any ideas?

**Comment by tznind** [comment]
What docker image are you using? Are you starting the binary in container
or just deploying into it?

Normally I think you would want to just deploy the code in the container
then use docker exec with `-it` to run.  That would ensure console output
from container goes to your local shell.

**Reply by thomaslazar** [reply]
Was using the mcr.microsoft.com/devcontainers/dotnet:7.0 image in the devcontainer.json file. 

Honestly? I just wanted to go in VS Code and press F5 and start debugging/running the code. 

Currently mucking around with building in the container, but then running it locally on my Mac in PowerShell as it seems to not work great on the normal terminal app. 

I was just fiddling around with it and don't intend to do much with it besides a few proof of concepts.

**Reply by BDisp** [reply]
There isn't a straight way to do this automatically by pressing `F5` to run a `devcontainer.json`. If you look at the root of `Terminal.Gui` folder you will find a `devcontainer.json`. Normally when opening the `Terminal.Gui` on `VS Code` a popup will appear on the lower right of the screen asking "Reopen in Container" which will allow to be run on the container after build. Inside the container it's possible run by pressing F5. If you want to create another container, search for `docker` on the open/closed issues or pull request of the `Terminal.Gui` repo. The UICatalog folder also have a `Dockerfile` with `opensuse` where you can see how to create and install the necessary dependencies for it could run. Normally right click it you can select `Build image...` but then you to run it as @tznind said.
